src/s2/s2region_term_indexer_benchmark.cpp

```diff
@@ -0,0 +1,67 @@
+#include "s2region_term_indexer.h"
+#include "s2testing.h"
+#include <benchmark/benchmark.h>
+
+static void BM_OldApiGetIndexTerms(benchmark::State &state) {
+  S2RegionTermIndexer indexer;
+  benchmark::DoNotOptimize(indexer);
+  auto const count = state.range(0);
+  for (auto _ : state) {
+    for (int64_t i = 0; i < count; ++i) {
+      auto value = indexer.GetIndexTerms(S2Testing::RandomPoint(), {});
```

Comment:
You're only testing the `S2Point` overload. Is that the only one you care about?

For that one, you can easily exactly compute the size you need and `reserve` it. I will add this and include it in the next PR.

What are the numbers when you do that?

Reply:
I wrote answer in other PR.

Benchmarks became same if you make reserve, not because reserve make same it still N allocations instead of 1.
But because it's microbenchmark, and in real code allocator have other jobs

> You're only testing the S2Point overload.

Not, but it's pretty same if we have covering (I added overload only for such functions)

Reply:
https://github.com/google/s2geometry/pull/303#discussion_r1066756513

Why is so much resistance about this?

I don't think it's needed for other API, like java or golang, because they care about performance a more little than cpp code